diff --git a/src/misc/lv_event.c b/src/misc/lv_event.c
--- a/src/misc/lv_event.c
+++ b/src/misc/lv_event.c
@@ -317,6 +317,8 @@ uint32_t lv_event_register_id(void)

 void lv_event_mark_deleted(void * target)
 {
+    if(target == NULL) return;
+
     lv_event_t * e = event_head;

     while(e) {
diff --git a/src/core/lv_obj.c b/src/core/lv_obj.c
--- a/src/core/lv_obj.c
+++ b/src/core/lv_obj.c
@@ -602,6 +602,7 @@ static void lv_obj_constructor(const lv_obj_class_t * class_p, lv_obj_t * obj)
 static void lv_obj_destructor(const lv_obj_class_t * class_p, lv_obj_t * obj)
 {
     LV_UNUSED(class_p);
+    if(obj == NULL) return;

     lv_event_mark_deleted(obj);

diff --git a/src/core/lv_obj_tree.c b/src/core/lv_obj_tree.c
--- a/src/core/lv_obj_tree.c
+++ b/src/core/lv_obj_tree.c
@@ -667,7 +667,7 @@ static void obj_indev_reset(lv_indev_t * indev, lv_obj_t * obj)

 static void obj_delete_core(lv_obj_t * obj)
 {
-    if(obj->is_deleting)
+    if(obj == NULL || obj->is_deleting)
         return;

     obj->is_deleting = true;
